src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -272,25 +272,21 @@ private protected static CodeActionResolveData CreateCodeActionResolveData(strin
             => new CodeActionResolveData(uniqueIdentifier, location.Range, CreateTextDocumentIdentifier(location.Uri));
 
         /// <summary>
-        /// Creates a solution with a document.
+        /// Creates an LSP server backed by a workspace instance with a solution containing the markup.
         /// </summary>
-        /// <returns>the solution and the annotated ranges in the document.</returns>
-        protected TestWorkspace CreateTestWorkspace(string markup, out Dictionary<string, IList<LSP.Location>> locations)
-            => CreateTestWorkspace(new string[] { markup }, out locations, LanguageNames.CSharp);
+        protected TestLspServer CreateTestLspServer(string markup, out Dictionary<string, IList<LSP.Location>> locations)
```

Comment:
Previously, all tests manually created the requestdispatcher and executionqueue themselves by importing the mef exports for them.  Tests that wanted to re-use the execution queue had to manually keep it around (the dispatcher always used the same mef instances).  This has to change to support non-mef handlers.

Tests shouldn't really need to know or worry much about the execution queue / dispatcher, so I put the TestWorkspace, queue and request dispatcher inside the TestLspServer.  Tests still have access to the workspace to set text buffers, options, etc, but the dispatcher and queue now only have test accessors and are otherwise 'invisible' to the tests.  A ton of the changes in this PR are just mechanical changes around all the usages of this.

I think we can potentially do more cleanup around this and limit the access tests have to the TestWorkspace, but I'm not going to do that here.